Sources/Subs-Db-postgresql.php

```diff
@@ -202,6 +202,13 @@ function smf_db_replacement__callback($matches)
 				smf_db_error_backtrace('Wrong value type sent to the database. Date expected. (' . $matches[2] . ')', '', E_USER_ERROR, __FILE__, __LINE__);
 		break;
 
+		case 'time':
+			if (preg_match('~^([0-1]?\d|2[0-3]):([0-5]\d):([0-5]\d)$~', $replacement, $time_matches) === 1)
+				return sprintf('\'%02d:%02d:%02d\'', $time_matches[1], $time_matches[2], $time_matches[3]);
```

Comment:
I don't understand your suggestion. The `smf_db_replacement__callback()` function validates and formats the values using PHP before anything is passed to the database. If you want to use SQL functions for that job, that would need to happen at a different, later stage in the process.